streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractKeyValueStoreTest.java

```diff
@@ -389,4 +391,17 @@ public void shouldNotThrowConcurrentModificationException() {
 
         assertEquals(new KeyValue<>(0, "zero"), results.next());
     }
+
+    @Test
+    public void shouldNotThrowInvalidRangeExceptionWithNegativeFromKey() {
+        LogCaptureAppender.setClassLoggerToDebug(InMemoryWindowStore.class);
+        final LogCaptureAppender appender = LogCaptureAppender.createAndRegister();
+
+        store.range(-1, 1);
```

Comment:
Could you add a check to verify that the returned iterator is empty. Something along the lines of `assertThat(iterator.hasNext(), is(false))`?

Could you also add a test for a range query where the start key is equal to the end key? Such a unit test ensures correct behaviour for this special case.    

nit: I would rename the test to `shouldReturnEmptyIteratorForRangeQueryWithInvalidKeyRange`. Correct me, if I am wrong, but I think the empty iterator and the invalid key range are the points here, not the negative starting key. I would even change the range from (-1, 1) to (5, 3). It took me a bit to understand why (-1, 1) is an invalid range. 

These comments apply also to the unit tests below.

Reply:
Ack Re: verify returned iterator is empty, add unit tests for equal start/end keys

Regarding your third point, this patch is mostly aimed at the bug in  [https://issues.apache.org/jira/browse/KAFKA-8159] 

which went undiscovered for a while because there were no tests of range queries with a negative key. I actually think it's fair to say we make no guarantees about what will happen if your app makes an invalid query; however we definitely shouldn't crash on what appears to be a valid query range (ie [-1,1]), which is the key point here

Reply:
Fair enough